todo-app/src/app/home/home.component.ts

```diff
@@ -1,69 +1,50 @@
 import { Component, OnInit } from '@angular/core';
-import { MserviceService } from '../mservice.service';
+import { HomeService } from './home.service';
+
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
-  styleUrls: ['./home.component.scss']
+  styleUrls: ['./home.component.scss'],
+  providers: [HomeService]
 })
 export class HomeComponent implements OnInit {
+  public state$: any;
+  public state: any = [];
+  public chosenTask: any = {};
 
-  constructor(private myservice: MserviceService) { }
-  listTask: any = [];
-  chosenTask: any = {};
-  ngOnInit(): void {
-    this.myservice.getData().subscribe((data) => {
+  constructor(private homeService: HomeService) { }
 
-      this.listTask = data;
-      console.log(this.listTask);
-    });
-    console.log(this.listTask);
+  ngOnInit(): void {
+    this.homeService.getData();
+    this.state$ = this.homeService.state$
+    this.state$.subscribe((state: any) => this.state = state);
```

Comment:
tham khảo
`  get state$(): Observable<IMG62202State> {
    return this.service.state$;
  }
`